packages/next-frontend/src/components/create-campaign/form/pool-step/picker/index.tsx

```diff
@@ -0,0 +1,155 @@
+import type { Erc20Token, Pool } from "@metrom-xyz/sdk";
+import { TextInput } from "@/src/ui/text-input";
+import AutoSizer from "react-virtualized-auto-sizer";
+import { FixedSizeList } from "react-window";
+import { useCallback, useState, type ChangeEvent, useMemo } from "react";
+import { SearchIcon } from "@/src/assets/search-icon";
+import { useTranslations } from "next-intl";
+import type { CampaignPayload } from "@/src/types";
+import { useDebounce } from "@/src/hooks/useDebounce";
+import { usePools } from "@/src/hooks/usePools";
+import { Chip } from "@/src/ui/chip/chip";
+import { useBaseTokens } from "@/src/hooks/useBaseTokens";
+import { RemoteLogo } from "@/src/ui/remote-logo";
+import { Typography } from "@/src/ui/typography";
+import { filterPools } from "@/utils/campaigns";
+import { Row } from "./row";
+
+import styles from "./styles.module.css";
+
+interface PoolPickerProps {
+    value?: Pool;
+    amm?: CampaignPayload["amm"];
+    onChange: (pool: Pool) => void;
+}
+
+export function PoolPicker({ value, amm, onChange }: PoolPickerProps) {
+    const t = useTranslations("new_campaign.form.pool");
+    const [search, setSearch] = useState("");
+    const [debouncedSearch, setDebouncedSearch] = useState(search);
+    const [baseTokenFilter, setBaseTokenFilter] = useState<Erc20Token>();
+
+    const baseTokens = useBaseTokens();
+    const { pools, loading, error } = usePools({ ammSlug: amm?.slug });
+
+    const filteredPools = useMemo(
+        () => filterPools(pools, baseTokenFilter?.address || debouncedSearch),
+        [baseTokenFilter?.address, debouncedSearch, pools],
+    );
+
+    useDebounce(() => {
+        setDebouncedSearch(search);
+    }, 300);
+
+    const getBaseTokenChangeHandler = useCallback(
+        (token: Erc20Token) => {
+            return () => {
+                if (token.address === baseTokenFilter?.address)
+                    setBaseTokenFilter(undefined);
+                else setBaseTokenFilter(token);
+            };
+        },
+        [baseTokenFilter?.address],
+    );
+
+    const handleSearchOnChange = useCallback(
```

Comment:
Same as above, no dependencies

Reply:
Fixed in 37d1fe105c5501a9d1d968990c6856b08fd8738d.